src/Resources/config/services.yaml

```diff
@@ -23,8 +23,23 @@ services:
     MonsieurBiz\SyliusNoCommercePlugin\Menu\ShopAccountMenuListener:
         tags:
             - { name: kernel.event_listener, event: sylius.menu.shop.account, priority: -10000 }
+                
+    MonsieurBiz\SyliusNoCommercePlugin\EventListener\NoShopListener:
+        tags:
+            - { name: kernel.event_listener, event: kernel.request, priority: -10000 }
 
     MonsieurBiz\SyliusNoCommercePlugin\Model\ConfigInterface: '@MonsieurBiz\SyliusNoCommercePlugin\Model\Config'
     MonsieurBiz\SyliusNoCommercePlugin\Model\Config:
         arguments:
             - '%monsieurbiz_sylius_nocommerce.config%'
+
+    Symfony\Bundle\SecurityBundle\Security\FirewallMap: '@security.firewall.map'
+
+    monsieurbiz.nocommerce.firewall.registry:
+        class: MonsieurBiz\SyliusNoCommercePlugin\Firewall\Registry
+
+    MonsieurBiz\SyliusNoCommercePlugin\Firewall\RegistryInterface: '@monsieurbiz.nocommerce.firewall.registry'
+    
+    MonsieurBiz\SyliusNoCommercePlugin\Form\Type\Settings\NoCommerceType:
+        arguments:
+            - '@monsieurbiz.nocommerce.firewall.registry'
```

Comment:
This is useless if you use autowiring, which is the case by default for all settings I think.

Reply:
i can't remove this without an error